src/classes/frDonation.cls

```diff
@@ -62,76 +62,180 @@ public class frDonation extends frModel implements frSyncable {
     public Boolean sync() {
         Boolean result = false;
         Map<String, Object> request = getRequestBody();
-        o = new Opportunity();
-        applyMappings(o, request);
-        setOpportunityStage(o, String.valueOf(request.get('status')));
-        
-        String funraiseId = getFunraiseId();       
-        String supporterFunraiseId = String.valueOf(request.get('donorId'));        
-        if (String.isNotBlank(supporterFunraiseId)) {
-            List<Contact> contacts = [SELECT Id, AccountId FROM Contact WHERE fr_ID__c = :supporterFunraiseId];
-            if (contacts.size() > 0) {
-                Contact supporterContact = contacts.get(0);
-                o.fr_Donor__c = supporterContact.Id;
-                if(frUtil.hasNPSP()){
-                    o.put('npsp__Primary_Contact__c', supporterContact.Id);
-                    o.AccountId = supporterContact.AccountId;
+        String frId = getFunraiseId();
+        SObject gt = null;
+        for (String key : request.keySet()) {
+            Object value = request.get(key);
+            System.debug('Key: ' + key + ', Value: ' + value);
+        }
+        // Gift Transaction Split
+        if (Schema.getGlobalDescribe().containsKey('gifttransaction')) {
+            //add a check for a dupe transaction?
+            String objectName = 'gifttransaction';  
+            List<SObject> gtList = Database.query('SELECT Id, Name FROM GiftTransaction WHERE fr_ID__c = :frId');
+            if(gtList.isEmpty()) {
+                gt = Schema.getGlobalDescribe().get(objectName).newSObject();
+            } 
+            else {
+                gt = gtList.get(0);
+            }
+            gt.put('Name', String.valueOf(request.get('name')));
+            gt.put('OriginalAmount', (Decimal) request.get('amount'));
+            gt.put('PaymentMethod', String.valueOf(request.get('paymentMethodType')));
+            gt.put('TransactionDueDate', System.today());
+            /*GiftTransaction gt = new GiftTransaction();
+            gt.Name = String.valueOf(request.get('name'));
+            gt.OriginalAmount = (Decimal) request.get('amount');
+            gt.PaymentMethod = String.valueOf(request.get('paymentMethodType'));
+            gt.TransactionDueDate = System.today();*/
+            applyMappings(gt, request);
+            System.debug('out of mappings');
+            //setOpportunityStage(o, String.valueOf(request.get('status'))); TODO: Have a similar function but for gt.Status
```

Comment:
Leaving a comment here to keep track of this TODO for mapping Funraise Status -> GT Status